mitiq/lre/inference/multivariate_richardson.py

```diff
@@ -0,0 +1,170 @@
+            for base, exp in zip(scale_factors, exponent):
+                # raise scale factor value by the exponent dict value
+                evaluated_terms.append(base**exp)
+            # multiply both elements in the list to create an evaluated
+            # monomial term
+            sample_matrix[i, j] = np.prod(evaluated_terms)
+
+    return sample_matrix
+
+
+def linear_combination_coefficients(
+    input_circuit: Circuit,
+    degree: int,
+    fold_multiplier: int,
+    num_chunks: Optional[int] = None,
+) -> list[float]:
+    r"""
+    Defines the function to find the linear combination coefficients from the
+    sample matrix as required for multivariate extrapolation (defined in
+    :cite:`Russo_2024_LRE`).
+
+    Args:
+        input_circuit: Circuit to be scaled.
+        degree: Degree of the multivariate polynomial.
+        fold_multiplier: Scaling gap required by unitary folding.
+        num_chunks: Number of desired approximately equal chunks. When the
+            number of chunks is the same as the layers in the input circuit,
+            the input circuit is unchanged.
+
+    Returns:
+        List of the evaluated monomial basis terms using the scale factor
+            vectors.
+    """
+    input_sample_matrix = sample_matrix(
+        input_circuit, degree, fold_multiplier, num_chunks
+    )
+    try:
+        det = np.linalg.det(input_sample_matrix)
+    except RuntimeWarning:  # pragma: no cover
+        # taken from https://stackoverflow.com/a/19317237
+        warnings.warn(  # pragma: no cover
+            "To account for overflow error, required determinant of "
+            + "large sample matrix is calculated through "
+            + "`np.linalg.slogdet`."
+        )
+        sign, logdet = np.linalg.slogdet(  # pragma: no cover
+            input_sample_matrix
+        )
+        det = sign * np.exp(logdet)  # pragma: no cover
+
+    if np.isinf(det):
+        raise ValueError(  # pragma: no cover
+            "Determinant of sample matrix cannot be calculated as "
+            + "the matrix is too large. Consider chunking your"
+            + " input circuit. "
+        )
+    assert det != 0.0
+
+    coeff_list = []
+    mat_row, mat_cols = input_sample_matrix.shape
+    assert mat_row == mat_cols
+    # replace a row of the sample matrix with [1, 0, 0, .., 0]
+    repl_row = np.array([[1] + [0] * (mat_cols - 1)])
+    for i in range(mat_row):
+        if i == 0:  # first row
+            new_mat = np.concatenate(
+                (repl_row, input_sample_matrix[1:]), axis=0
+            )
+        elif i == mat_row - 1:  # last row
+            new_mat = np.concatenate(
+                (input_sample_matrix[:i], repl_row), axis=0
+            )
+        else:
+            frst_sl = np.concatenate(
+                (input_sample_matrix[:i], repl_row), axis=0
+            )
+            sec_sl = input_sample_matrix[i + 1 :]
+            new_mat = np.concatenate((frst_sl, sec_sl), axis=0)
```

Comment:
@natestemen @vprusso Does this sound like a better alternative to the [previous version](https://github.com/unitaryfund/mitiq/blob/7d8868b94fd7404e552f6703b335893e13dc17fc/mitiq/lre/inference/multivariate_richardson.py#L192) where I was copying the sample matrix? 

Both of you pointed out memory issues with using `.copy()` on a numpy array.

Reply:
This seems okay from my perspective!

Reply:
Discussed with @natestemen to revert back to using `.copy()` on a numpy array because `np.concatenate` might have the same memory issues as `.copy()`. 

The code is more readable with `.copy()`. Will add this as a todo as discussed in https://github.com/unitaryfund/mitiq/pull/2447#discussion_r1729114577